ui/components/app/import-token/token-search/token-search.component.js

```diff
@@ -56,8 +73,15 @@ export default function TokenSearch({
 
   const clear = () => {
     setSearchQuery('');
+    setSearchResults([]);
   };
 
+  useEffect(() => {
+    // When the token network filter changes, reset the search query.
+    handleSearch('');
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isTokenNetworkFilterEqualCurrentNetwork]);
```

Comment:
This is more of a nit, but it would be great if we can remove this lint ignore here.

Out of curiosity, whey aren't we calling `clear()` here? Do we want to retain search results after network switch? It might seem odd to clear the search query, but retain the results.